Lib/Src/CUartCom.cpp

```diff
@@ -10,67 +10,79 @@
 
 #include "CUartCom.h"
 
+CUartCom *CUartCom::sp_UART_1 = NULL;
+CUartCom *CUartCom::sp_UART_2 = NULL;
+
 /**
  * @brief Construct UART communication object.
- *
  * @param p_huart Pointer to UART hardware control register structure.
+ * @param name Name of the instance
+ * @note If no USART_DE Pin is needed, the BREATHING Pin will turn on when UART
+ * is sending data
  */
-CUartCom::CUartCom(UART_HandleTypeDef *p_huart) : mp_huart(p_huart)
+CUartCom::CUartCom(UART_HandleTypeDef *p_huart, const std::string name)
+    : IComChannel(name),
+      mp_huart(p_huart),
+      m_uart_de_pin(BREATHING_GPIO_Port, BREATHING_Pin)
 {
     // Check mp_huart is not null
     if (!mp_huart)
     {
-        // Hang application with a toggling ALARM Pin
-        while (1)
-        {
-            // TODO: cannot use alarm pin. No space for it due to one less pin
-            // available on this MCU. HAL_GPIO_TogglePin(ALARM_GPIO_Port,
-            // ALARM_Pin);
-            HAL_Delay(200);
-        }
+        Error_Handler();
     }
 }
 
 /**
  * @brief Construct UART communication object with half-duplex mode.
- *
  * @param p_huart Pointer to UART hardware control register structure.
+ * @param name Name of the CUartCom Instance
  * @param uart_de_port Pointer to GPIO port.
  * @param uart_de_pin GPIO pin.
  */
 CUartCom::CUartCom(UART_HandleTypeDef *p_huart,
```

Comment:
This is going to fail if this class is instantiated as a global variable on the heap instead of on the stack as part of a controller. It is best to not rely on `p_huart` being defined. 
If you don't understand the differences I am talking about here, ask me and I'll explain.